### PR TITLE
further enhancements to front-endiness of loading GH issues and updating submissions

### PR DESCRIPTION
**Pull request by RobertAKARobin**

### CODE REVIEW

**Comment by RobertAKARobin** [comment]
Further enhances the front-endiness of loading issues.

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -13,3 +13,15 @@
 //= require jquery
 //= require jquery_ujs
 //= require_tree .
+
+
+function update(url, object){
```
Comment (jshawl) [inline app/assets/javascripts/application.js]: would be useful to add a callback here as function argument, which is invoked in promise, or return promise

File: app/assets/javascripts/assignments.js [hunk 2]
```diff
@@ -35,3 +35,42 @@ $(function(){
     })
   }
 })
+
```
Comment (jshawl) [inline app/assets/javascripts/assignments.js]: does the below code make the above code obsolete?

File: app/models/assignment.rb [hunk 3]
```diff
@@ -50,4 +50,12 @@ def api_repo_issues_url
     url += "/issues?state=all"
   end
```
Comment (jshawl) [inline app/models/assignment.rb]: does the below code make the above code obsolete?

File: app/views/assignments/show.html.erb [hunk 4]
```diff
@@ -5,13 +5,12 @@
 
 <section>
 <h3>Submissions</h3>
-<p>
-  <%= @submissions_percent_complete %>% complete.
-</p>
+<p><%= percent_of(@submissions, 2) %>% complete</p>
+<button onclick="loadIssues('<%= assignment_issues_path(@assignment) %>')">Count issues</button>
```
Comment (jshawl) [inline app/views/assignments/show.html.erb]: does this need to be an AJAX call? why not just count from the controller?

**Comment by jshawl** [comment]
I'm a little bummed about having to mark something complete here, and then as well on github. There is no visual representation of whether an issue is present or not.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (RobertAKARobin) [inline app/assets/javascripts/application.js]: Good call

File: app/models/assignment.rb (hunk 3 above)
Comment (RobertAKARobin) [inline app/models/assignment.rb]: No; `#issues_url` just returns the URL at which the issues can be found

**Comment by RobertAKARobin** [comment]
There is! This introduces a "Count issues" button, which, upon clicking, replaces the text of all the "View issues" links with the number of "open" and "closed" issues for each user (see image).

screen shot 2015-10-28 at 9 57 18 am

The advantage of doing it this way is that the user may not need to see the counts of all the issues, so the AJAX call only fires if they need it. Marking a submission as "complete" and closing issues are still separate processes, but I think that's good because otherwise a student could mark all their homework as "complete" simply by opening an issue and closing it, and they may have multiple issues/PRs on one repo.

**Comment by jshawl** [comment]
I'm getting an off by one error

**Comment by RobertAKARobin** [comment]
OK, will investigate

File: app/assets/javascripts/assignments.js (hunk 2 above)
Comment (RobertAKARobin) [inline app/assets/javascripts/assignments.js]: I guess so, but I felt nervous and so didn't want to remove it.

**Comment by RobertAKARobin** [comment]
@jshawl Responded to your comments; thanks! I think this should fix it.

**Comment by jshawl** [comment]
last two are showing issues, but only the last one has an issue

**Pull request merged**